tf_encrypted/keras/layers/pooling_test.py

```diff
@@ -0,0 +1,58 @@
+# pylint: disable=missing-docstring
+import unittest
+
+import numpy as np
+import tensorflow as tf
+
+import tf_encrypted as tfe
+from tf_encrypted.keras.testing_utils import agreement_test
+
+np.random.seed(42)
+
+
+class TestPooling2d(unittest.TestCase):
+
+  def setUp(self):
+    tf.reset_default_graph()
+
+  def test_maxpooling2d_valid(self):
+    self._core_maxpooling2d(padding='valid')
```

Comment:
is it worth adding tests for alternative paddings?

Reply:
applies to both max and avg

Reply:
I would also add a test using the `channels_first` data_format

Reply:
In fact, keras only support channels_last on cpu. However I think we could leave this option for TFE. 

```
 File "/Users/yanndupis/Documents/dropoutlabs/tfe_clean_copy/tf-encrypted/tf_encrypted/keras/testing_utils.py", line 42, in agreement_test
    y = tf_layer(x)
  File "/anaconda3/envs/py35_int64/lib/python3.5/site-packages/tensorflow/python/keras/engine/base_layer.py", line 554, in __call__
    outputs = self.call(inputs, *args, **kwargs)
  File "/anaconda3/envs/py35_int64/lib/python3.5/site-packages/tensorflow/python/keras/layers/pooling.py", line 256, in call
    data_format=conv_utils.convert_data_format(self.data_format, 4))
  File "/anaconda3/envs/py35_int64/lib/python3.5/site-packages/tensorflow/python/ops/nn_ops.py", line 2718, in avg_pool
    name=name)
  File "/anaconda3/envs/py35_int64/lib/python3.5/site-packages/tensorflow/python/ops/gen_nn_ops.py", line 93, in avg_pool
    data_format=data_format, name=name)
  File "/anaconda3/envs/py35_int64/lib/python3.5/site-packages/tensorflow/python/framework/op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "/anaconda3/envs/py35_int64/lib/python3.5/site-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "/anaconda3/envs/py35_int64/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 3300, in create_op
    op_def=op_def)
  File "/anaconda3/envs/py35_int64/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 1801, in __init__
    self._traceback = tf_stack.extract_stack()

InvalidArgumentError (see above for traceback): Default AvgPoolingOp only supports NHWC on device type CPU
	 [[node average_pooling2d/AvgPool (defined at /Users/yanndupis/Documents/dropoutlabs/tfe_clean_copy/tf-encrypted/tf_encrypted/keras/testing_utils.py:42) ]]
```

Reply:
makes sense -- I'd actually add one more test for each of these, where the stride and pool size differ from each other.  I suggest something like pool_size=2 and stride=1

Reply:
Sounds good. I just added this test.